flask_alchemydumps/__init__.py

```diff
@@ -108,13 +108,27 @@ def restore(date_id):
 
             # restore to the db
             db = alchemy.db()
+            sequence = 0
+            table_name = mapped_class.__tablename__
+            sequence_name = '{}_id_seq'.format(table_name)
             for row in alchemy.parse_data(contents):
                 try:
                     db.session.merge(row)
+                    # updating sequence
+                    sequence = alchemy.get_next_sequence(
+                        sequence_name)
                     db.session.commit()
                 except (IntegrityError, InvalidRequestError):
                     db.session.rollback()
                     fails.append(row)
+            max_id = db.session.execute(
```

Comment:
@cuducos Sometime max id from the table is different than the max no of elements, so validating that here.